engines/bastion_katello/app/assets/javascripts/bastion_katello/content-hosts/bulk/views/content-hosts-bulk-system-purpose-modal.html

```diff
@@ -0,0 +1,77 @@
+<div data-extend-template="components/views/bst-modal.html">
+  <h4 data-block="modal-header" translate>Content Host System Purpose</h4>
+
+  <div data-block="modal-body">
+    <h4 translate>Assign System Purpose:</h4>
+
+    <div class="row">
+      <div class="col-sm-12">
+        <div bst-global-notification></div>
+      </div>
+    </div>
+
+    <form name="contentHostContentForm" class="form" ng-hide="content.workingMode">
+      <div>
+        <label translate>Service Level:</label>
+
+        <select type="select"
+                ng-options="item for item in defaultServiceLevels"
+                ng-model="selectedServiceLevels">
+        </select>
+      <br /><br />
+
+        <label translate>Usage Type:</label>
+
+        <select type="select"
+                readonly="denied('edit_hosts', host)"
+                ng-options="item for item in defaultUsages"
+                ng-model="selectedRoles">
+        </select>
+        <br /><br />
+
+        <label translate>Role:</label>
+
+        <select type="select"
+                readonly="denied('edit_hosts', host)"
+                ng-options="item for item in defaultRoles"
+                ng-model="selectedUsages">
+        </select>
+        <br /><br />
+
+        <label ng-hide="purposeAddonsCount === 0" translate>Add ons:</label>
+        <select multiple ng-multiple="true"
+                ng-hide="purposeAddonsCount === 0"
+                readonly="denied('edit_hosts', host)"
+                ng-options="item for item in purposeAddonsList"
+                ng-model="selected.addons">
+        </select>
+        <br /><br />
+
+      </div>
+
+      <div bst-alert="info" ng-show="showConfirm">
+        <span translate>
+          Are you sure you want to set the System Purpose values on the content host(s) selected?
```

Comment:
Forgot to say this before, but it would be super helpful if this could be a dynamic message like
```
Set System Purpose values on 422 selected content hosts?
```
```suggestion
          Set System Purpose values on {{ hostIds.length }} selected content hosts?
```

Reply:
I could not get that to work, using the console I was able to find it like this `hostIds.included.ids.length` but putting it in the view did not work so just made a variable to reference the count. Not sure if I am doing something wrong or just how our page behaves. Tried `{{ table.numSelected }}` as well but no luck

Reply:
it may just be my lack of knowledge of Angular.  I'll see if i can figure it out